File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number: int):
    return sum(range(1, number + 1))**2


def sum_of_squares(number: int):
    return sum(map(lambda x: x**2, range(1, number + 1)))


def difference_of_squares(number: int):
    return abs(square_of_sum(number) - sum_of_squares(number))

File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sum
    1.upto(@number).sum ** 2
  end

  def sum_of_squares
    1.upto(@number).map{ |number| number ** 2}.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end